### ISSUE TITLE
buildah push  to dir: creates .tar file which is not in tar format

### ISSUE THREAD

**Issue by adelton** [projectatomic/buildah#481]
**`buildah push`  to `dir:` creates `.tar` file which is not in tar format**

**Steps to reproduce the issue:**
1. `./buildah from registry.fedoraproject.org/fedora:rawhide`
2. `mkdir fedora-rawhide`
3. `./buildah push registry.fedoraproject.org/fedora:rawhide dir:fedora-rawhide`
4. `ls -l fedora-rawhide`
5. `tar tvf fedora-rawhide/feec4342ed84f31082461f08a31662af85d61177bb83936ed5cc5a0ceb5755d4.tar`

**Describe the results you received:**

```
# ./buildah from registry.fedoraproject.org/fedora:rawhide
Getting image source signatures
Copying blob sha256:7ecc8a985be6848a7ee66b80217c168de6e79dcbda3c01ee36a45367c3a3a6bc
 86.23 MiB / 86.23 MiB [====================================================] 9s
Copying config sha256:feec4342ed84f31082461f08a31662af85d61177bb83936ed5cc5a0ceb5755d4
 1.27 KiB / 1.27 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
fedora-working-container
# mkdir fedora-rawhide
# ./buildah push registry.fedoraproject.org/fedora:rawhide dir:fedora-rawhide
Getting image source signatures
Copying blob sha256:02a71b4f9b794b507208390fac92f88382f10ca62be5073397a82cde9c6a586f
 251.90 MiB / 251.90 MiB [==================================================] 6s
Copying config sha256:feec4342ed84f31082461f08a31662af85d61177bb83936ed5cc5a0ceb5755d4
 1.27 KiB / 1.27 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
# ls -l fedora-rawhide
total 257964
-rw-r--r--. 1 root root 264140800 Feb 16 20:12 02a71b4f9b794b507208390fac92f88382f10ca62be5073397a82cde9c6a586f.tar
-rw-r--r--. 1 root root      1299 Feb 16 20:12 feec4342ed84f31082461f08a31662af85d61177bb83936ed5cc5a0ceb5755d4.tar
-rw-r--r--. 1 root root       430 Feb 16 20:12 manifest.json
-rw-r--r--. 1 root root        33 Feb 16 20:12 version
# tar tvf fedora-rawhide/feec4342ed84f31082461f08a31662af85d61177bb83936ed5cc5a0ceb5755d4.tar 
tar: This does not look like a tar archive
tar: Skipping to next header
tar: Exiting with failure status due to previous errors
```

**Describe the results you expected:**

If `feec4342ed84f31082461f08a31662af85d61177bb83936ed5cc5a0ceb5755d4.tar` has `.tar` suffix, I'd expected it to be in tar format.

Since it seems to be JSON, it should be named `feec4342ed84f31082461f08a31662af85d61177bb83936ed5cc5a0ceb5755d4.json`.

**Output of `rpm -q buildah` or `apt list buildah`:**

I'm running `buildah` built from master (09e7ddf5441f1734e5d920c354b8d2f6d1ac21a9)

**Output of `buildah version`:**

```
# ./buildah version
Version:       0.12
Go Version:    go1.9.3
Image Spec:    1.0.0
Runtime Spec:  1.0.0
Git Commit:    09e7ddf
Built:         Fri Feb 16 19:00:00 2018
OS/Arch:       linux/amd64
```

**Output of `cat /etc/*release`:**

```
# cat /etc/*release
Fedora release 27 (Twenty Seven)
NAME=Fedora
VERSION="27 (Twenty Seven)"
ID=fedora
VERSION_ID=27
PRETTY_NAME="Fedora 27 (Twenty Seven)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:27"
HOME_URL="https://fedoraproject.org/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=27
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=27
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
Fedora release 27 (Twenty Seven)
Fedora release 27 (Twenty Seven)
```

**Output of `uname -a`:**

```
Linux redacted.example.com 4.14.18-300.fc27.x86_64 #1 SMP Thu Feb 8 01:35:31 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
# storage.conf is the configuration file for all tools
# that share the containers/storage libraries
# See man 5 containers-storage.conf for more information

# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "overlay"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers.
size = ""

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
override_kernel_check = "true"
```

**Comment by rhatdan** [comment]
Urvashi can you take a look at this?

**Comment by umohnani8** [comment]
Fixed in #489
